src/optimizer/statistics/stats_calculator.cpp

```diff
@@ -162,6 +162,20 @@ void StatsCalculator::Visit(const LogicalDelete *op) {
   }
 }
 
+void StatsCalculator::Visit(const LogicalCreateIndex *op) {
+  auto *root_group = context_->GetMemo().GetGroupByID(gexpr_->GetGroupID());
+
+  // Compute selectivity at the first time
+  if (root_group->GetNumRows() == Group::UNINITIALIZED_NUM_ROWS) {
+    const auto latched_table_stats_reference = context_->GetStatsStorage()->GetTableStats(
+        op->GetDatabaseOid(), op->GetTableOid(), context_->GetCatalogAccessor());
```

Comment:
The `OptimizerContext` has pointers to both the `TransactionContext` and the `CatalogAcessor` which seem like more appropriate places from which fetch the `DatabaseOid` for storing stats.